test/e2e/lifecycle/recreate_node.go

```diff
@@ -0,0 +1,120 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+var _ = SIGDescribe("Recreate [Disruptive] [Feature:Recreate]", func() {
+	f := framework.NewDefaultFramework("recreate")
+	var originalNodes []v1.Node
+	var originalPodNames []string
+	var ps *testutils.PodStore
+	systemNamespace := metav1.NamespaceSystem
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("gce", "gke")
+		var err error
+		ps, err = testutils.NewPodStore(f.ClientSet, systemNamespace, labels.Everything(), fields.Everything())
+		numNodes, err := framework.NumberOfRegisteredNodes(f.ClientSet)
+		Expect(err).NotTo(HaveOccurred())
+		originalNodes, err = framework.CheckNodesReady(f.ClientSet, numNodes, framework.NodeReadyInitialTimeout)
+		Expect(err).NotTo(HaveOccurred())
+
+		framework.Logf("Got the following nodes before recreate %v", nodeNames(originalNodes))
+
+		ps, err = testutils.NewPodStore(f.ClientSet, systemNamespace, labels.Everything(), fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+		allPods := ps.List()
+
+		originalPods := filterIrrelevantPods(allPods)
+		originalPodNames = make([]string, len(originalPods))
+		for i, p := range originalPods {
+			originalPodNames[i] = p.ObjectMeta.Name
+		}
+
+		if !framework.CheckPodsRunningReadyOrSucceeded(f.ClientSet, systemNamespace, originalPodNames, framework.PodReadyBeforeTimeout) {
+			printStatusAndLogsForNotReadyPods(f.ClientSet, systemNamespace, originalPodNames, originalPods)
+			framework.Failf("At least one pod wasn't running and ready or succeeded at test start.")
+		}
+
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			// Make that addon/system pods are running, so dump
+			// events for the kube-system namespace on failures
+			By(fmt.Sprintf("Collecting events from namespace %q.", systemNamespace))
+			events, err := f.ClientSet.CoreV1().Events(systemNamespace).List(metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, e := range events.Items {
+				framework.Logf("event for %v: %v %v: %v", e.InvolvedObject.Name, e.Source, e.Reason, e.Message)
+			}
+		}
+		if framework.ProviderIs("gke") {
```

Comment:
Does this test need to ssh into node? I think this extra wait for tunnel dropping is only needed if we need to run some ssh commands on test nodes.

Reply:
Ah ok, thanks for the info. Removed since this test doesn't ssh into any node.